cmd/tailscale/callbacks.go

```diff
@@ -96,12 +96,27 @@ func Java_com_tailscale_ipn_IPNService_connect(env *C.JNIEnv, this C.jobject) {
 	onConnect <- jni.NewGlobalRef(jenv, jni.Object(this))
 }
 
+//export Java_com_tailscale_ipn_IPNService_directConnect
+func Java_com_tailscale_ipn_IPNService_directConnect(env *C.JNIEnv, this C.jobject) {
+	requestBackend(ConnectEvent{Enable: true})
```

Comment:
I'm concerned these three new methods will be a race, the existing `connect()` and `disconnect()` methods use a Go channel to serialize requests. These calls straight into IPN.

Presumably you started by trying to call `connect()`, what happened that prompted adding `directConnect()` ?

Reply:
That doesn't work, `directConnect()` simulates the command ran that the UI makes when toggling the switch to join the VPN - `connect()` in that context doesn't do anything